plugins/action_send_notification/src/sendNotification.ts

```diff
@@ -0,0 +1,127 @@
+import { ActionQueueStatus } from '../../../src/generated/graphql'
+import { ActionPluginType } from '../../types'
+import databaseMethods from './databaseMethods'
+import path from 'path'
+import nodemailer from 'nodemailer'
+import marked from 'marked'
+import config from '../config.json'
+import { Attachment } from 'nodemailer/lib/mailer'
+import { getFilePath } from '../../../src/components/files/fileHandler'
+import { ActionApplicationData } from '../../../src/types'
+
+const sendNotification: ActionPluginType = async ({ parameters, applicationData, DBConnect }) => {
+  const db = databaseMethods(DBConnect)
+  const { host, port, secure, user, defaultFromName, defaultFromEmail } = config
+  const {
+    userId = applicationData?.userId,
+    email = applicationData?.email,
+    fromName = defaultFromName,
+    fromEmail = defaultFromEmail,
+    subject,
+    message,
+    attachments = [],
+    sendEmail = true,
+  } = parameters
+
+  const {
+    environmentData: { appRootFolder, filesFolder },
+  } = applicationData as ActionApplicationData
+
+  const transporter = nodemailer.createTransport({
+    host,
+    port,
+    secure,
+    auth: {
+      user,
+      pass: process.env.SMTP_PASSWORD,
+    },
+  })
+
+  try {
+    const emailAddressString = stringifyEmailRecipientsList(email)
+
+    // Create notification database record
+    console.log('Creating notification...')
+    let notificationResult
+    notificationResult = await db.createNotification({
+      userId,
+      applicationId: applicationData?.applicationId,
+      emailAddressString,
+      subject,
+      message,
+      attachments,
+    })
+
+    // Send email
+    if (sendEmail) {
+      console.log('Sending email...')
+      // Note: Using "any" type as imported @types defintions is incorrect, doesn't recognise some fields on "SentMessageInfo" type
+      const emailResult: any = await transporter.sendMail({
+        from: `${fromName} <${fromEmail}>`,
+        to: emailAddressString,
+        subject,
+        text: message,
+        html: marked(message),
```

Comment:
Curious what does `marked` package does?

Reply:
It's the Markdown parser. Same core as the one we use in front-end, but the front-end one has a React wrapper.